lib/widget/helpers/widgets.dart

```diff
@@ -78,4 +82,61 @@ class BoxWrapper extends StatelessWidget {
                     ]));
   }
 
+}
+
+
+class GradientBorder extends BoxBorder {
+  const GradientBorder({this.gradient, this.width});
+
+  final Gradient? gradient;
+
+  final double? width;
+
+  @override
+  BorderSide get bottom => BorderSide.none;
+
+  @override
+  BorderSide get top => BorderSide.none;
+
+  @override
+  EdgeInsetsGeometry get dimensions => EdgeInsets.all(width!);
+
+  @override
+  bool get isUniform => true;
+
+  @override
+  void paint(
+    Canvas canvas,
+    Rect rect, {
+    TextDirection? textDirection,
+    BoxShape shape = BoxShape.rectangle,
+    BorderRadius? borderRadius,
+  }) {
+     if (borderRadius != null) {
+          _paintRRect(canvas, rect, borderRadius);
+          return;
+        }
+        _paintRect(canvas, rect);
+  }
+
+  void _paintRect(Canvas canvas, Rect rect) {
+    canvas.drawRect(rect.deflate(width! / 2), _getPaint(rect));
+  }
+
+  void _paintRRect(Canvas canvas, Rect rect, BorderRadius borderRadius) {
+    final rrect = borderRadius.toRRect(rect).deflate(width! / 2);
+    canvas.drawRRect(rrect, _getPaint(rect));
+  }
+
+  @override
+  ShapeBorder scale(double t) {
+    return this;
+  }
+
+  Paint _getPaint(Rect rect) {
+    return Paint()
+      ..strokeWidth = width!
```

Comment:
This is very important and I can't stress enough many times. Never force ! unless the editor needs help. Your params are optional above and this will throw an error.